aviatrix/resource_aviatrix_kubernetes_cluster.go

```diff
@@ -0,0 +1,285 @@
+				Description: "Map of tags.",
+			},
+		},
+	}
+}
+
+func parseArn(arn string) (accountId string, region string, name string, err error) {
+	if !strings.HasPrefix(arn, "arn:") {
+		err = errors.New("ARN must start with 'arn:'")
+		return
+	}
+	sections := strings.SplitN(arn, ":", 6)
+	if len(sections) != 6 {
+		err = errors.New("ARN must have 6 sections")
+	}
+	return sections[4], sections[3], sections[5], nil
+}
+
+func marshalKubernetesClusterInput(d *schema.ResourceData) (*goaviatrix.KubernetesCluster, error) {
+	var clusterId string
+	arn, ok := d.GetOk("arn")
+	if ok {
+		accountId, region, name, err := parseArn(arn.(string))
+		if err != nil {
+			return nil, err
+		}
+		clusterId = fmt.Sprintf("%s/%s/%s", accountId, region, name)
+	} else {
+		clusterId = d.Get("cluster_id").(string)
+	}
+
+	kubernetesCluster := &goaviatrix.KubernetesCluster{
+		ClusterId: clusterId,
+		Credential: &goaviatrix.KubernetesCredential{
+			UseCspCredentials: d.Get("use_csp_credentials").(bool),
+			KubeConfig:        d.Get("kube_config").(string),
+		},
+	}
+
+	if accountName, ok := d.GetOk("account_name"); ok {
+		resource := goaviatrix.ClusterResource{
+			AccountName: accountName.(string),
+			AccountId:   d.Get("account_id").(string),
+			Name:        d.Get("name").(string),
+			VpcId:       d.Get("vpc_id").(string),
+			Region:      d.Get("region").(string),
+			Version:     d.Get("version").(string),
+			Platform:    d.Get("platform").(string),
+			Public:      d.Get("public").(bool),
+			NetworkMode: d.Get("network_mode").(string),
+		}
+		if project, ok := d.GetOk("project"); ok {
+			resource.Project = project.(string)
+		}
+		if compartment, ok := d.GetOk("compartment"); ok {
+			resource.Compartment = compartment.(string)
+		}
+		if tags, ok := d.Get("tags").(map[string]interface{}); ok {
+			for key, value := range tags {
+				resource.Tags = append(resource.Tags, goaviatrix.Tag{
+					Key:   key,
+					Value: value.(string),
+				})
+			}
+		}
+		kubernetesCluster.Resource = &resource
+	}
+
+	return kubernetesCluster, nil
+}
+
+func resourceAviatrixKubernetesClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*goaviatrix.Client)
+
+	kubernetesCluster, err := marshalKubernetesClusterInput(d)
+	if err != nil {
+		return diag.Errorf("failed to marshal kubernetes cluster: %v", err)
+	}
+	if err := client.CreateKubernetesCluster(ctx, kubernetesCluster); err != nil {
+		return diag.Errorf("failed to create kubernetes cluster: %s", err)
+	}
+
+	d.SetId(kubernetesCluster.Id)
+	return resourceAviatrixKubernetesClusterRead(ctx, d, meta)
+}
+
+func resourceAviatrixKubernetesClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*goaviatrix.Client)
+
+	kubernetesCluster, err := client.GetKubernetesCluster(ctx, d.Id())
+	if err != nil {
+		return diag.Errorf("failed to read kubernetes credential: %s", err)
+	}
```

Comment:
Should you handle NotFound specifically and set
```
d.SetId("")
return nil
```

Reply:
Thanks! That's better.
In that case Terraform figures that the object was deleted and recreates it.